### PR TITLE
Throw if UseUrls specifies HTTPS or path base (#1519)

### PR DESCRIPTION
**Pull request by cesarbs**
#1519

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs [hunk 1]
```diff
@@ -164,9 +164,14 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                     {
                         var parsedAddress = ServerAddress.FromUrl(address);
 
+                        if (parsedAddress.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase))
+                        {
+                            throw new InvalidOperationException($"HTTPS addresses are not supported. Use {nameof(KestrelServerOptions)}.{nameof(KestrelServerOptions.Listen)}() to configure an HTTPS endpoint.");
+                        }
+
                         if (!string.IsNullOrEmpty(parsedAddress.PathBase))
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: What happens if it's just '/'?
Where does it throw? This just logs.

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: Oops 😄

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: There's a `ServerAddress` test that checks that just '/' is not set as path base: https://github.com/aspnet/KestrelHttpServer/blob/632780dd16729298933f96796e6a969f3470d810/test/Microsoft.AspNetCore.Server.KestrelTests/ServerAddressTests.cs#L42

File: src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs [hunk 2]
```diff
@@ -164,9 +164,14 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                     {
                         var parsedAddress = ServerAddress.FromUrl(address);
 
+                        if (parsedAddress.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase))
+                        {
+                            throw new InvalidOperationException($"HTTPS addresses are not supported. Use {nameof(KestrelServerOptions)}.{nameof(KestrelServerOptions.Listen)}() to configure an HTTPS endpoint.");
```
Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: I don't like the phrasing of "HTTPS addresses are not supported..."

I would change this to "HTTPS endpoints may only be configured using {nameof(KestrelServerOptions)}.{nameof(KestrelServerOptions.Listen)}()".

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: may?

Comment (halter73) [inline src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs]: can, must, whatever.

**Comment by cesarbs** [comment]
@halter73 I've changed the path base message too, for consistency.

**Comment by natemcmaster** [comment]
Does this incidentally fix https://github.com/aspnet/KestrelHttpServer/issues/1296 too?

**Comment by cesarbs** [comment]
@natemcmaster No. See my updated comment there.

**Comment by cesarbs** [comment]
Updated.

**Pull request merged**